test/unit_test.c

```diff
@@ -2205,7 +2205,7 @@ static void eh6(struct mg_connection *c, int ev, void *ev_data, void *fn_data) {
   (void) c, (void) ev_data;
 }
 
-#if defined(__arm__) || defined(__riscv)
```

Comment:
Sure. It's to differentiate between arm and armhf.
Arm build only requires dummy function.
Amrhf requires the BSD sockets (it would compile but not run with the dummy function).
I took advantage of Makefile which already defines MG_ARCH_NEWLIB for arm build (not defined for armhf)
which is why i changed this check.

Agree it's not perfectly clear, but __arm__ is defined for both and therefore can't be used here.
Difference is armhf now runs. Arm was meant as build only test as far as i could see. Runtime doesn't work
yet.

Reply:
Got it, that construction is doomed to fail on several platforms... if the need is to define a dummy send() when there is no OS with socket support I would check for that... MG_ARCH_NEWLIB would be indirect, as it defines MG_ENABLE_SOCKET=0. I would check for MG_ENABLE_SOCKET==0 and remove the `__riscv` check

Reply:
Ok, include fixed.